src/components/ItemFilter/FilterDialog.js

```diff
@@ -20,21 +18,73 @@ import {
     DIMENSION_ID_ORGUNIT,
 } from '@dhis2/analytics'
 
-class FilterDialog extends Component {
-    onConfirm = id => () => this.props.onConfirm(id)
+import { acAddItemFilter, acRemoveItemFilter } from '../../actions/itemFilters'
+import { sGetItemFiltersRoot } from '../../reducers/itemFilters'
+
+const FilterDialog = (
+    {
+        displayNameProperty,
+        dimension,
+        initiallySelectedItems,
+        addItemFilter,
+        removeItemFilter,
+        onClose,
+    },
+    context
+) => {
+    const [filters, setFilters] = useState(initiallySelectedItems)
 
-    renderDialogContent() {
-        const { displayNameProperty, dimension, selectedItems } = this.props
-        const dialogId = dimension.id
+    const onSelectItems = ({ dimensionId, items }) => {
+        setFilters({ [dimensionId]: items })
+    }
 
+    const onDeselectItems = ({ dimensionId, itemIdsToRemove }) => {
+        const oldList = filters[dimensionId] || []
+        const newList = oldList.filter(
+            item => !itemIdsToRemove.includes(item.id)
+        )
+
+        const list = newList.length ? newList : []
```

Comment:
Is this necessary?
If newList is empty, it's already `[]`

Reply:
Not sure. I just copied this code over from FilterSelector. Will check.

Reply:
Removed the unneeded line